src/backend/replication/walproposer.c

```diff
@@ -2164,6 +2139,14 @@ AsyncReadMessage(Safekeeper *sk, AcceptorProposerMessage *anymsg)
 			msg->hs.ts = pq_getmsgint64_le(&s);
 			msg->hs.xmin.value = pq_getmsgint64_le(&s);
 			msg->hs.catalog_xmin.value = pq_getmsgint64_le(&s);
+			if (buf_size > APPENDRESPONSE_FIXEDPART_SIZE)
+			{
+				StringInfoData s;
```

Comment:
We probably can reuse existing StringInfoData s, and replace `buf_size > APPENDRESPONSE_FIXEDPART_SIZE` with `s.len - s.cursor > 0`. Not sure how to keep AppendResponse backwards compatible in the future though.